src/components/data/data-utils.js

```diff
@@ -939,6 +940,111 @@ export const MultipleEntityEditor = ({ entityType, entityNames, attributeNames,
   ])
 }
 
+export const AddEntityModal = ({ workspaceId: { namespace, name }, entityType, attributeNames, entityTypes, onDismiss, onSuccess }) => {
+  const [entityName, setEntityName] = useState('')
+  const [entityNameInputTouched, setEntityNameInputTouched] = useState(false)
+  const [takenNames, setTakenNames] = useState([])
+
+  // Default all attribute values to empty strings
+  const [attributeValues, setAttributeValues] = useState(_.fromPairs(_.map(attributeName => [attributeName, ''], attributeNames)))
+
+  const [isBusy, setIsBusy] = useState(false)
```

Comment:
Nit: From other components, my eyes are used to easily locating state thanks to having chunks like:

```suggestion
  // Default all attribute values to empty strings
  const [attributeValues, setAttributeValues] = useState(_.fromPairs(_.map(attributeName => [attributeName, ''], attributeNames)))
  const [isBusy, setIsBusy] = useState(false)
```